manager/src/main/java/com/redhat/service/bridge/manager/workers/resources/AbstractWorker.java

```diff
@@ -0,0 +1,137 @@
+package com.redhat.service.bridge.manager.workers.resources;
+
+import java.time.ZonedDateTime;
+import java.util.Objects;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.service.bridge.infra.models.dto.ManagedResourceStatus;
+import com.redhat.service.bridge.manager.models.ManagedResource;
+import com.redhat.service.bridge.manager.models.Work;
+import com.redhat.service.bridge.manager.workers.WorkManagerImpl;
+import com.redhat.service.bridge.manager.workers.Worker;
+
+import io.quarkus.hibernate.orm.panache.PanacheRepositoryBase;
+
+public abstract class AbstractWorker<T extends ManagedResource> implements Worker<T> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractWorker.class);
+
+    private static final Set<ManagedResourceStatus> PROVISIONING_STARTED = Set.of(ManagedResourceStatus.ACCEPTED, ManagedResourceStatus.PROVISIONING);
+    private static final Set<ManagedResourceStatus> DEPROVISIONING_STARTED = Set.of(ManagedResourceStatus.DEPROVISION, ManagedResourceStatus.DELETING);
+
+    private static final Set<ManagedResourceStatus> PROVISIONING_COMPLETED = Set.of(ManagedResourceStatus.READY, ManagedResourceStatus.FAILED);
+    private static final Set<ManagedResourceStatus> DEPROVISIONING_COMPLETED = Set.of(ManagedResourceStatus.DELETED, ManagedResourceStatus.FAILED);
+
+    @ConfigProperty(name = "event-bridge.resources.worker.max-retries")
+    int maxRetries;
+
+    @ConfigProperty(name = "event-bridge.resources.workers.timeout-seconds")
+    int timeoutSeconds;
+
+    @Inject
+    WorkManagerImpl workManager;
+
+    @Override
+    public T handleWork(Work work) {
+        T managedResource = load(work);
+        if (Objects.isNull(managedResource)) {
+            //Work has been scheduled but cannot be found. Something (horribly) wrong has happened.
+            String message = String.format("Resource of type '%s' with id '%s' no longer exists in the database.", work.getType(), work.getManagedResourceId());
+            throw new IllegalStateException(message);
+        }
+
+        // Fail when we've had enough
+        if (areRetriesExceeded(work)) {
+            LOGGER.error(
+                    "Max retry attempts exceeded trying to create dependencies for '{}' [{}].",
+                    managedResource.getName(),
+                    managedResource.getId());
+            managedResource.setStatus(ManagedResourceStatus.FAILED);
+            persist(managedResource);
+            return managedResource;
```

Comment:
looking at the impl 
```java
        boolean complete = false;
        if (managedResource.getStatus() == ManagedResourceStatus.ACCEPTED) {
            complete = createDependencies(managedResource).getDependencyStatus().isReady();
        } else if (managedResource.getStatus() == ManagedResourceStatus.DEPROVISION) {
            complete = deleteDependencies(managedResource).getDependencyStatus().isDeleted();
        }

        if (complete) {
            workManager.complete(work);
        } else {
            workManager.schedule(managedResource);
```
I think we are not adding any delay in the scheduling: in the case of connectors, for example, we are going to hammer the fleet manager of MC with a lot of requests until they return that the connector is ready. 
I think we should have a delay instead